src/scripts/danish-legal/create_completeness_detection.py

```diff
@@ -0,0 +1,299 @@
+)
+
+# Hardcoded mapping to the indices in `contract_sections`
+# for each category.
+REQUIRED_CATEGORIES_TO_INDICES = {
+    "Partner/Identitet": [0],
+    "Arbejdstid": [6],
+    "Ansvarsområde": [2],
+    "Ansættelse": [1],
+    "Ferie/Fravær": [7, 8, 10, 11, 12, 13],
+    "Ophør/Opsigelse": [17],
+    "Løn/Vederlag": [3, 4, 5],
+    "Klausuler": [18],
+    "Personale vilkår": [9],
+}
+
+
+@click.command()
+@click.argument("contract_path", type=click.Path(exists=True, path_type=Path))
+@click.option(
+    "--num-samples",
+    "-n",
+    default=10,
+    help=f"Number of samples to generate (default: {10})",
+)
+def main(contract_path: Path, num_samples: int) -> None:
+    """Create the legal completeness detection dataset and upload it to the HF Hub.
+
+    This script modifies the input contract to create a dataset of samples
+    with complete and incomplete contracts.
+
+    Args:
+        contract_path: Path to the contract PDF file
+        num_samples: Number of samples to generate
+    """
+    # Load and process contract
+    logger.info(f"Processing contract from {contract_path}")
+    if not contract_path.exists():
+        click.echo(f"Error: Contract file {contract_path} does not exist", err=True)
+        return
+
+    converter = DocumentConverter()
+    result = converter.convert(str(contract_path))
+    markdown = result.document.export_to_markdown()
+
+    # Parse contract into sections
+    contract_sections = parse_contract_into_sections(markdown)
+
+    # Get non-required indices
+    not_required_indices = [
+        i
+        for i in range(len(contract_sections))
+        if not any(i in indices for indices in REQUIRED_CATEGORIES_TO_INDICES.values())
+    ]
+
+    # Generate samples
+    logger.info(f"Generating {num_samples} unique samples...")
+    samples = generate_unique_samples(
+        contract_sections=contract_sections,
+        not_required_indices=not_required_indices,
+        num_samples=num_samples,
+    )
+
+    df = pd.DataFrame(samples)
+    dataset = DatasetDict(test=Dataset.from_pandas(df, split=Split.TEST))
+
+    # Remove existing dataset if it exists
+    dataset_id = "EuroEval/legal-completeness-detection"
+    try:
+        api = HfApi()
+        api.delete_repo(dataset_id, repo_type="dataset")
+    except HTTPError:
+        pass
+
+    # Upload dataset
+    dataset.push_to_hub(dataset_id, private=True)
+
+
+def parse_contract_into_sections(markdown_text: str) -> List[str]:
+    """Parse markdown into sections, excluding subsections with bullet points.
+
+    Args:
+        markdown_text: The markdown text of the contract
+
+    Returns:
+        A list of sections
+    """
+    sections = re.split(
+        r"(?=^## \d+\.\s+(?!.*\n\n-))", markdown_text, flags=re.MULTILINE
+    )
+    return [section.strip() for section in sections if section.strip()]
+
+
+def generate_unique_samples(
+    contract_sections: List[str], not_required_indices: List[int], num_samples: int
```

Comment:
No need to use the `List` class from `typing` anymore, we can now just use the primitives 🙂 


```suggestion
    contract_sections: list[str], not_required_indices: list[int], num_samples: int
```

Same thing goes for `Dict`/`dict` and `Tuple`/`tuple`.